### PR TITLE
describe fairness and how that relates to priorities

### LINKED ISSUE #1819: Define fairness
From the TSVART review, Bos says:

> §13. Fairness
> 
> T#13a) Please define fair. Seriously.
> 
> A prerequisite question is, "What parameters does a server scheduler
> manipulate?" The text implies the server can only control the order in which it
> starts a response to each request, and whether responses start while other
> responses are in progress or wait for their completion. If so, I'm not sure
> what fairness could mean.
> 
> Presumably a server can also determine the relative rate at which it sends
> different streams. And it could stall a stream to allow another absolute
> priority. In this context, fairness might mean instantaneously equal rates. But
> that's not fair if the objects are of different sizes.
> 
> So we genuinely do need to see a definition of what fairness means here.

We should address this.

### LINKED ISSUE #1820: Why does the IETF have anything to say about scheduling across mutliple connections?
As part of the TSVART review, Boc said

>  T#13b) Why not make scheduling decisions across different clients?
> 
>    As a general guideline, a server SHOULD NOT use priority information
>    for making scheduling decisions across multiple connections, unless
>    it knows that those connections originate from the same client.
> 
> Why does the IETF have anything to say about this? It's surely an operator
> policy decision.

This seems like a fair point and we likely shouldn't say anything. We can remove this statement entirely but it is normative text so it'll be a change.

### PR DESCRIPTION
**Pull request by kazuho**
As discussed in https://github.com/httpwg/http-extensions/issues/1819#issuecomment-987199413 and https://github.com/httpwg/http-extensions/issues/1820#issuecomment-987196961, describe fairness and how that relates to priorities.

Closes #1819, closes #1820.

### CODE REVIEW

File: draft-ietf-httpbis-priority.md [hunk 1]
```diff
@@ -813,15 +813,19 @@ might use the relative priority of streams when choosing probe data.
...
 # Fairness {#fairness}
 
-As a general guideline, a server SHOULD NOT use priority information for making
-scheduling decisions across multiple connections, unless it knows that those
-connections originate from the same client. Due to this, priority information
-conveyed over a non-coalesced HTTP connection (e.g., HTTP/1.1) might go unused.
-
-The remainder of this section discusses scenarios where unfairness is
-problematic and presents possible mitigations, or where unfairness is desirable.
+Typically, HTTP stacks depend on the underlying transport to maintain fairness
```
Comment (LPardue) [inline draft-ietf-httpbis-priority.md:816]: Not a huge fan of "stack" here. Could we use implementation, application, or something different?

**Review by LPardue — APPROVED** [review]
LGTM modulo the term "stacks"

File: draft-ietf-httpbis-priority.md (hunk 1 above)
Comment (kazuho) [reply draft-ietf-httpbis-priority.md]: :+1: Went with "implementation."

File: draft-ietf-httpbis-priority.md [hunk 2]
```diff
@@ -813,15 +813,19 @@ might use the relative priority of streams when choosing probe data.
...
-scheduling decisions across multiple connections, unless it knows that those
-connections originate from the same client. Due to this, priority information
-conveyed over a non-coalesced HTTP connection (e.g., HTTP/1.1) might go unused.
-
-The remainder of this section discusses scenarios where unfairness is
-problematic and presents possible mitigations, or where unfairness is desirable.
+Typically, HTTP implementations depend on the underlying transport to maintain
+fairness between connections competing for bandwidth. When HTTP requests are
+forwarded through intermediaries, progress made by each connection originating
+from end clients can become different over time, depending on how intermediaries
```
Comment (ioggstream) [inline draft-ietf-httpbis-priority.md:819]:
```suggestion
from user agents can become different over time, depending on how intermediaries
```

File: draft-ietf-httpbis-priority.md [hunk 3]
```diff
@@ -813,15 +813,19 @@ might use the relative priority of streams when choosing probe data.
...
+coalesce or split requests into backend connections. This unfairness can expand
+if priority signals are used. {{coalescing}} and {{h1-backends}} discuss
+mitigations against this expansion of unfairness.
```
Comment (ioggstream) [inline draft-ietf-httpbis-priority.md:822]:
```suggestion
mitigations against this amplification of unfairness.
```

File: draft-ietf-httpbis-priority.md [hunk 4]
```diff
@@ -851,7 +855,7 @@ header fields:
 * Forwarded {{?FORWARDED=RFC7239}}, X-Forwarded-For
 * Via (see {{Section 7.6.3 of HTTP}})
 
-## HTTP/1.x Back Ends
+## HTTP/1.x Back Ends {#h1-backends}
 
 It is common for CDN infrastructure to support different HTTP versions on the
```
Comment (ioggstream) [inline draft-ietf-httpbis-priority.md:860]:
```suggestion
It is common for CDN infrastructures to support different HTTP versions on the
```

File: draft-ietf-httpbis-priority.md [hunk 5]
```diff
@@ -864,7 +868,7 @@ Back end servers SHOULD only schedule based on client priority information where
 that information can be scoped to individual end clients. Authentication and
```
Comment (ioggstream) [inline draft-ietf-httpbis-priority.md:868]:
```suggestion
that information can be scoped to individual user agents. Authentication and
```

**Review by ioggstream — COMMENTED** [review]
Minor edits

File: draft-ietf-httpbis-priority.md (hunk 2 above)
Comment (LPardue) [reply draft-ietf-httpbis-priority.md:819]: When talking about intermediaries, I prefer the term end clients

File: draft-ietf-httpbis-priority.md (hunk 3 above)
Comment (LPardue) [reply draft-ietf-httpbis-priority.md:822]: using expansion is consistent with the previous sentence

File: draft-ietf-httpbis-priority.md (hunk 4 above)
Comment (LPardue) [reply draft-ietf-httpbis-priority.md:860]: the current text ok

**Pull request merged**